pkg/config/tmpl/rulesconfig.go

```diff
@@ -275,6 +308,24 @@ func (rc *RulesConfig) Merge(other TemplateConfig) error {
 					return err
 				}
 			}
+
+			// Set the Scope field on the rule if the condition has a scope meta value
+			// This is ugly because the structure of the rules is ugly.
+			if scope, exists := otherRC.meta["scope"]; exists && scope != "" {
+				// Merge scope values: if either is "span", result is "span", otherwise "trace"
+				currentScope := "trace" // default scope
+				if ruleIndex < len(sampler.RulesBasedSampler.Rules) {
+					if existingScope := sampler.RulesBasedSampler.Rules[ruleIndex].Scope; existingScope != "" {
+						currentScope = existingScope
+					}
+				}
+
+				mergedScope := mergeScopeValues(currentScope, scope)
+				if err := setMemberValue(fmt.Sprintf("RulesBasedSampler.Rules.%d.Scope", ruleIndex), sampler, mergedScope); err != nil {
+					return err
+				}
+			}
```

Comment:
Is this the same code as above? If so a shared function would be nice.

Reply:
Similar, but not identical. I played with it a bit, but there are enough differences that it didn't feel worth it to try to combine them (the function would have several arguments).